UnitsNet/CustomCode/UnitClasses/Density.extra.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright © 2007 by Initial Force AS.  All rights reserved.
+// https://github.com/InitialForce/UnitsNet
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+namespace UnitsNet
+{
+    public partial struct Density
+    {
+        public static Mass operator *(Density density, Volume volume)
+        {
+            return Mass.FromKilograms(density.KilogramsPerCubicMeter * volume.CubicMeters);
+        }
+
+        public static Mass operator *(Volume volume, Density density)
```

Comment:
This applies to all units, but I think we should establish a consistent pattern for where operator overloading is added. The same applies to unit tests of cross-unit conversions.

```public static Mass operator *(Volume volume, Density density)```

Should this reside in Volume, Density or Mass?

For me, I find the mest intuitive that the left-hand operator determines this, in this case `Volume`, since that is what we are "acting" on. We have `Volume`, and we want to turn it into something else. The main drawback is that for multiplication, where one often adds an overload of the same calculation, but with `Volume` as right-hand operator will be moved to `Density.exta.cs` instead, so it is not clear when reading whether both have been implemented or not, but I find that drawback less bad than violating the pattern by grouping them both in `Volume.extra.cs` or both in `Density.extra.cs`.

Thoughts?

I'm fine cleaning up this in a separate PR in order to get this merged in sooner.